backend/src/src-components/src/main/java/com/siemens/sw360/components/ComponentHandler.java

```diff
@@ -365,12 +367,27 @@ public RequestStatus removeAttachmentFromRelease(String releaseId, User user, St
                 return RequestStatus.FAILURE;
             }
             attachments.remove(attachment);
+            if (attachment.getAttachmentType() == AttachmentType.CLEARING_REPORT){
+                autosetClearingState(release, attachments);
+            }
             return updateRelease(release, user);
         } else {
             return RequestStatus.SUCCESS;
         }
     }
 
+    private void autosetClearingState(Release release, Set<Attachment> attachments) {
+        if (attachments.stream().anyMatch(att -> att.getAttachmentType()== AttachmentType.CLEARING_REPORT)){
+            if (attachments.stream().anyMatch(att -> att.getAttachmentType()==AttachmentType.CLEARING_REPORT && att.getCheckStatus() == CheckStatus.ACCEPTED)){
```

Comment:
I think, it would be nice to refactor these conditions into two predicates `hasClearingReport` and `hasAcceptedClearingReport` (or `hasApprovedClearingReport`).